game-app/game-core/src/main/java/games/strategy/engine/ClientFileSystemHelper.java

```diff
@@ -38,19 +40,32 @@ public static Path getRootFolder() {
     }
   }
 
-  private static Path getCodeSourceFolder() throws IOException {
-    final ApplicationContext applicationContext = Services.loadAny(ApplicationContext.class);
-    final @Nullable CodeSource codeSource =
-        applicationContext.getMainClass().getProtectionDomain().getCodeSource();
-    if (codeSource == null) {
-      throw new IOException("code source is not available");
-    }
+  public static void setCodeSourceFolder(final Path sourceFolder) {
+    codeSourceLocation = sourceFolder;
+  }
 
-    final Path codeSourceLocation;
-    try {
-      codeSourceLocation = Path.of(codeSource.getLocation().toURI());
-    } catch (final URISyntaxException e) {
-      throw new IOException("code source location URI is malformed", e);
+  /**
+   * TripleA stores two folders, one for user content that survives between game installs, and a
+   * second that contains binaries. This method returns the 'user folder', which contains maps and
+   * save games.
+   *
+   * @return Folder that contains binaries. This folder
+   *     would contain e.g. resources that are not specific to a particular game.
+   */
+  public static Path getCodeSourceFolder() throws IOException {
+    if (codeSourceLocation == null) {
+      final ApplicationContext applicationContext = Services.loadAny(ApplicationContext.class);
+      final @Nullable CodeSource codeSource =
+          applicationContext.getMainClass().getProtectionDomain().getCodeSource();
+      if (codeSource == null) {
+        throw new IOException("code source is not available");
+      }
+
+      try {
+        codeSourceLocation = Path.of(codeSource.getLocation().toURI());
+      } catch (final URISyntaxException e) {
+        throw new IOException("code source location URI is malformed", e);
```

Comment:
nit, I don't think IOException is the correct type to throw. IllegalStateException perhaps is more fitting. 

Additionally, we could in addition print the actual malformed URI in the exception message to make it easier to debug what is wrong.

Reply:
Would you like me to add these corrections to my pull request? I didn't change this code

Reply:
If this is just moved, don't worry about it for right now. I'd certainly like to get this landed and let me know how I can help the most to do that. From my perspective we need to make sure all functionality is working well (to my knowledge the outstanding issue is selection of incorrect casualties). We'd done a few code passes for structure/design/maintenance and I'd like to get those wrapped up (and not add more).